cmake/StaticBuild.cmake

```diff
@@ -587,7 +587,7 @@ foreach(curl_arch ${curl_arches})
     --enable-crypto-auth --disable-ntlm-wb --disable-tls-srp --disable-unix-sockets --disable-cookies
     --enable-http-auth --enable-doh --disable-mime --enable-dateparse --disable-netrc --without-libidn2
     --disable-progress-meter --without-brotli --with-zlib=${DEPS_DESTDIR}
-    --without-ssl --without-schannel --without-secure-transport
+    --with-ssl --without-zstd --without-schannel --without-secure-transport
```

Comment:
The problem with this is that it's going to go looking for libssl to link again, and almost certainly will pick up the dynamic one, and thus we end up with a non-statically linked binary.

oxen-storage-server's cmake/StaticBuild.cmake does have openssl static build code that we should port over here to solve that.  (It's probably also worth diffing the curl build flags to these ones and see if anything got added there that is missing here).

Reply:
https://github.com/oxen-io/oxen-storage-server/blob/861638758ff626142446964f20d44e43b7055918/cmake/StaticBuild.cmake#L195 + the version/url/hash variables near the top of the file.

Reply:
Yep, have adapted over the changes from the script. Seems to be working